concordium-consensus/tests/scheduler/SchedulerTests/EncryptedTransfersTest.hs

```diff
@@ -71,949 +75,1365 @@ import Test.Hspec
+                        case Helpers.getResults $ Sch.ftAdded srTransactions of
+                            [(_, Types.TxSuccess{..})] ->
+                                assertEqual
+                                    "The correct events are produced"
+                                    [ Types.EncryptedAmountsRemoved
+                                        { earAccount = accountAddress0,
+                                          earUpToIndex = 2,
+                                          earNewAmount =
+                                            stpatdRemainingAmount
+                                                secToPubTransferData3,
+                                          earInputAmount = aggregatedEncryptedAmount8
+                                        },
+                                      Types.AmountAddedByDecryption
+                                        { aabdAccount = accountAddress0,
+                                          aabdAmount = 200
                                         }
-                                    thomasAccount
-                                    bs
-                            )
-                        ),
-                        ( Runner.TJSON
-                            { payload = Runner.EncryptedAmountTransfer alesAccount encryptedTransferData5,
-                              metadata = makeDummyHeader thomasAccount 2 100000,
-                              keys = [(0, [(0, thomasKP)])]
-                            },
-                            ( SuccessE
-                                [ EncryptedAmountsRemoved
-                                    { earAccount = thomasAccount,
-                                      earUpToIndex = 3,
-                                      earNewAmount = eatdRemainingAmount encryptedTransferData5,
-                                      earInputAmount = aggregatedEncryptedAmount5
-                                    },
-                                  NewEncryptedAmount
-                                    { neaAccount = alesAccount,
-                                      neaNewIndex = 1,
-                                      neaEncryptedAmount = eatdTransferAmount encryptedTransferData5
-                                    }
-                                ],
-                              \bs -> do
-                                checkEncryptedBalance
-                                    initialAccountEncryptedAmount
-                                        { _selfAmount = eatdRemainingAmount encryptedTransferData3,
-                                          _startIndex = 0,
-                                          _incomingEncryptedAmounts = incomingAmounts5A
+                                    ]
+                                    vrEvents
+                            _ -> assertFailure "Tenth transaction should succeed"
+                        doEncryptedBalanceAssertion
+                )
+            ]
+
+testCase1 ::
```

Comment:
Can you please put a sentence for each of these test cases. I know documentation hasn't been good, but since the entire file is refactored it should also be made more maintainable now that you had to understand some of it at least.

Reply:
I have added some comments, but the module documentation contains most of the details, so I only comment on the differences and refer to the module documentation for the rest.